Mage/src/main/java/mage/abilities/common/EntersBattlefieldOneOrMoreTriggeredAbility.java

```diff
@@ -40,37 +40,31 @@ private EntersBattlefieldOneOrMoreTriggeredAbility(final EntersBattlefieldOneOrM
 
     @Override
     public boolean checkEventType(GameEvent event, Game game) {
-        return event.getType() == GameEvent.EventType.ZONE_CHANGE_GROUP;
+        return event.getType() == GameEvent.EventType.ZONE_CHANGE_BATCH;
     }
 
     @Override
     public boolean checkTrigger(GameEvent event, Game game) {
-        ZoneChangeGroupEvent zEvent = (ZoneChangeGroupEvent) event;
+
         Player controller = game.getPlayer(this.controllerId);
-        if (zEvent.getToZone() != Zone.BATTLEFIELD || controller == null) {
+        if (controller == null) {
             return false;
         }
 
+        ZoneChangeBatchEvent zEvent = (ZoneChangeBatchEvent) event;
+        Stream<Permanent> enteringPermanents = zEvent.getEvents().stream()
+                .filter(z -> z.getToZone() == Zone.BATTLEFIELD)
+                .map(ZoneChangeEvent::getTarget)
+                .filter(Objects::nonNull)
+                .filter(permanent -> filterPermanent.match(permanent, this.controllerId, this, game));
+
         switch (this.targetController) {
             case YOU:
-                if (!controller.getId().equals(zEvent.getPlayerId())) {
-                    return false;
-                }
-                break;
+                return enteringPermanents.anyMatch(permanent -> permanent.getControllerId().equals(this.controllerId));
             case OPPONENT:
-                if (!controller.hasOpponent(zEvent.getPlayerId(), game)) {
-                    return false;
-                }
-                break;
+                return enteringPermanents.anyMatch(permanent -> controller.hasOpponent(this.controllerId, game));
```

Comment:
this line looks wrong, it should be checking `permanent.getControllerId()`. Looks like no ability currently uses TargetController.OPPONENT though.

also, should add a default case to the switch that throws an error (you can imagine a dev trying to use TargetController.ANY and it will silently never trigger)